### PR TITLE
[Driver][SYCL]Allowing to add C++ libraries to the link line with clang -fsycl

### PR DESCRIPTION
**Pull request by hchilama**
Icx with -fsycl option behaves as C++ compiler but it doesnot add C++ libraries at link time. So to have some alignment between compilation and link step we need icx to add C++ libraries if -fsycl option is specified.

### CODE REVIEW

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -1593,6 +1593,10 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
     }
   }
 
+  bool IsSYCL =
+         Args.hasFlag(options::OPT_fsycl, options::OPT_fno_sycl, false);
+  if(IsSYCL && CCCIsCC()) setDriverMode("g++");
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:1598]:
```suggestion
  if (IsSYCL && CCCIsCC())
    setDriverMode("g++");
```

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -1593,6 +1593,10 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
     }
   }
 
+  bool IsSYCL =
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:1596]: nit - This boolean isn't used anywhere else, you could pull the hasFlag check into the if statement

File: clang/test/Driver/sycl-offload.cpp [hunk 3]
```diff
@@ -160,3 +160,6 @@
 // RUN:   %clangxx -### -fsycl-device-only %s 2>&1 | FileCheck -check-prefixes=CHECK_BITFIELD_OPTION %s
 // CHECK_BITFIELD_OPTION: clang{{.*}} "-ffine-grained-bitfield-accesses"
 
+/// Check if the clang with fsycl adds C++ libraries to the link line
+//  RUN:  %clang -### -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK-FSYCL-WITH-CLANG %s
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-offload.cpp:164]:
```suggestion
//  RUN:  %clang -### -target x86_64-unknown-linux-gnu -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK-FSYCL-WITH-CLANG %s
```
Add the explicit target to behave like Linux when run in a Windows environment.

**Comment by hchilama** [comment]
Opened a new PR with updated fork: https://github.com/intel/llvm/pull/9896
Closing this as it doesn't have clean runs.

**Pull request closed without merging**